sync/src/block/downloader/body.rs

```diff
@@ -111,6 +115,7 @@ impl BodyDownloader {
                 break
             }
         }
+        self.downloaded.shrink_to_fit();
         self.targets.drain(0..result.len());
```

Comment:
Please shrink `targets` too.

Reply:
locally I shrank `targets`. In next force-push the change will be included.